community/kernel/src/main/java/org/neo4j/graphdb/PropertyContainer.java

```diff
@@ -126,4 +128,19 @@
      */
     // TODO: figure out concurrency semantics
     Iterable<String> getPropertyKeys();
+
+    /**
+     * Returns specified existing properties.
+     *
+     * @param keys the property keys to return
+     * @return specified properties on this property container
```

Comment:
We should ask users to not rely on the mutability of the returned map and document what will happen if empty/null array of names is given as a parameter